great_expectations/rule_based_profiler/types/attributed_resolved_metrics.py

```diff
@@ -51,7 +48,15 @@ def id(self) -> str:
 
     @property
     def attributed_metric_values(self) -> Optional[Dict[str, MetricValue]]:
-        return self.metric_values_by_batch_id
+        if self.metric_values_by_batch_id is None:
+            return None
+
+        batch_id: str
+        return {
+            batch_id: self.metric_values_by_batch_id.get(batch_id)
```

Comment:
This is a nullable return type right? Should we have `Optional[MetricValue]`? Or are we confident about the key existing (we should then use []).

Reply:
Looking at L58, I think we should just use `[]` because we know that we have the key for sure.

Reply:
@cdkini We may not -- if metric calculation fails (in an earlier PR, I made sure that we get a log message).  Thanks.